atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/priority/NextTableToSweepProviderImpl.java

```diff
@@ -91,19 +104,52 @@ public NextTableToSweepProviderImpl(KeyValueService kvs, SweepPriorityStore swee
         }
     }
 
+    private double getSweepPriorityLogging(SweepPriority oldPriority, SweepPriority newPriority) {
+        double priority = getSweepPriority(oldPriority, newPriority);
+        log.debug("Priority for table {} is {}",
+                LoggingArgs.tableRef(newPriority.tableRef()),
+                SafeArg.of("priority", priority));
+        return priority;
+    }
+
     private double getSweepPriority(SweepPriority oldPriority, SweepPriority newPriority) {
         if (AtlasDbConstants.hiddenTables.contains(newPriority.tableRef())) {
-            // Never sweep hidden tables
+            // Never sweep hidden tables.
             return 0.0;
         }
         if (!newPriority.lastSweepTimeMillis().isPresent()) {
-            // Highest priority if we've never swept it before
+            // Highest priority if we've never swept it before.
             return Double.MAX_VALUE;
         }
         if (oldPriority.writeCount() > newPriority.writeCount()) {
             // We just swept this, or it got truncated.
             return 0.0;
         }
+
+        if (StreamTableType.isStreamStoreValueTable(newPriority.tableRef())) {
+            return getStreamStorePriority(newPriority);
+        } else {
+            return getNonStreamStorePriority(oldPriority, newPriority);
+        }
+    }
+
+    private double getStreamStorePriority(SweepPriority newPriority) {
+        long millisSinceSweep = System.currentTimeMillis() - newPriority.lastSweepTimeMillis().getAsLong();
+
+        if (TimeUnit.DAYS.convert(millisSinceSweep, TimeUnit.MILLISECONDS)
+                < DAYS_TO_WAIT_BEFORE_SWEEPING_STREAM_STORE_VALUE_TABLE) {
+            return 0L;
+        }
+        if (newPriority.writeCount() > STREAM_STORE_VALUES_TO_SWEEP) {
+            return Double.MAX_VALUE;
+        }
```

Comment:
I've added logging so we can start to understand the range of values we see.  I'd like to move to not using MAX_DOUBLE and then selecting randomly based on weights to avoid starving any table.